src/libraries/PoolTickBitmap.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.26;
+
+import {IPoolManager} from "lib/v4-core/src/interfaces/IPoolManager.sol";
+import {PoolId} from "lib/v4-core/src/types/PoolId.sol";
+import {BitMath} from "@uniswap/v4-core/src/libraries/BitMath.sol";
+import {StateLibrary} from "@uniswap/v4-core/src/libraries/StateLibrary.sol";
+
+/// @title Packed tick initialized state library
+/// @notice Stores a packed mapping of tick index to its initialized state
+/// @dev The mapping uses int16 for keys since ticks are represented as int24 and there are 256 (2^8) values per word.
+library PoolTickBitmap {
+    using StateLibrary for IPoolManager;
+
+    /// @notice Computes the position in the mapping where the initialized bit for a tick lives
+    /// @param tick The tick for which to compute the position
+    /// @return wordPos The key in the mapping containing the word in which the bit is stored
+    /// @return bitPos The bit position in the word where the flag is stored
+    function position(int24 tick) private pure returns (int16 wordPos, uint8 bitPos) {
+        wordPos = int16(tick >> 8);
+        bitPos = uint8(uint24(tick % 256));
+    }
+
+    /// @notice Returns the next initialized tick contained in the same word (or adjacent word) as the tick that is either
+    /// to the left (less than or equal to) or right (greater than) of the given tick
+    /// @param poolId The pool id
+    /// @param tickSpacing the tick spacing of the pool
+    /// @param tick The starting tick
+    /// @param lte Whether to search for the next initialized tick to the left (less than or equal to the starting tick)
+    /// @return next The next initialized or uninitialized tick up to 256 ticks away from the current tick
+    /// @return initialized Whether the next tick is initialized, as the function only searches within up to 256 ticks
+    function nextInitializedTickWithinOneWord(
+        IPoolManager poolManager,
+        PoolId poolId,
+        int24 tickSpacing,
+        int24 tick,
+        bool lte
+    ) internal view returns (int24 next, bool initialized) {
```

Comment:
it bothers me that we need to re-implement this again in periphery. Here's a draft PR on core which makes the library usable by periphery

https://github.com/Uniswap/v4-core/pull/832